php/Blockonomics.php

```diff
@@ -238,4 +294,334 @@ public function testSetup()
+        }
+        return false;
+    }
+
+    // Check and update the crypto order or create a new order
+    public function process_order($order_id, $crypto){
+        $orders = get_option('blockonomics_orders');
+
+        $orders_by_id = $this->get_all_orders_by_id($orders, $order_id);
+        if ($orders_by_id) {
+            $this->check_for_pending_payment($orders_by_id, $order_id);
+
+            $waiting_order = $this->get_waiting_order($orders_by_id, $order_id, $crypto);
+            if ($waiting_order) {
+                $order = $waiting_order;
+            }else {
+                $order = $this->create_new_order($order_id, $crypto);
+            }
+        }else {
+            $order = $this->create_new_order($order_id, $crypto);
+        }
+        
+        $orders[$order_id][$order['address']] = $order;
+        update_option('blockonomics_orders', $orders);
+
+        $order['crypto'] = $this->getActiveCurrencies()[$crypto];
+        return $order;
+    }
+
+    // Get the order info by id and crypto
+    public function get_order_info($order_id, $crypto){
+        $order = $this->process_order($order_id, $crypto);
+        header("Content-Type: application/json");
+        exit(json_encode($order));
+    }
+
+    // Get the order info by crypto address
+    public function get_order_by_address($address){
+        $orders = get_option('blockonomics_orders');
+        foreach($orders as $id => $order){
+            if(isset($order[$address])){
+                $order_id = $id;
+                return $order[$address];
+            }
+        }
+        exit("Error: order not found");
+    }
+
+    // Check if the callback secret in the request matches
+    public function check_callback_secret($secret){
+        $callback_secret = get_option("blockonomics_callback_secret");
+        if ($callback_secret  && $callback_secret == $secret) {
+            return true;
+        }
+        exit("Error: secret does not match");
+    }
+
+    // Save the received payment info to the WooCommerce order
+    public function record_payment($value, $order, $wc_order){
+        update_post_meta($wc_order->get_id(), 'paid_'. $order['crypto'], $value/1.0e8);
+        update_post_meta($wc_order->get_id(), 'blockonomics_'. $order['crypto'] .'_txid', $order['txid']);
+        update_post_meta($wc_order->get_id(), 'expected_'. $order['crypto'], $order['satoshi']/1.0e8);
+    }
+
+    public function check_for_late_payment($status, $order, $wc_order){
+        if($status == 0){
+            $time_period = get_option("blockonomics_timeperiod", 10) * 60;
+            if (time() > $order['timestamp'] + $time_period) {
+                $minutes = (time() - $order['timestamp']) / 60;
+                $wc_order->add_order_note(__("Warning: Payment arrived after $minutes minutes. Received". $order['crypto'] ."may not match current ". $order['crypto'] ." price", 'blockonomics-bitcoin-payments'));
+            }
+        }
+    }
```

Comment:
Remove this. Theortically this is useful, but practically it confuses merchants a lot. And we haven't see anyone using/requesting this

Reply:
Removed in ac5506b